packages/shared/src/open-feature.ts

```diff
@@ -124,7 +124,15 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
 
     const emitters = this.getAssociatedEventEmitters(clientName);
 
-    if (typeof provider.initialize === 'function') {
+    // TODO: this can be removed after 1.0
+    if (typeof provider.initialize === 'function' && provider.status === undefined) {
+      const activeLogger = this._logger || console;
+      activeLogger.warn(
+        `Provider ${provider?.metadata?.name} implements 'initialize' but not 'status'. Please implement 'status'.`
+      );
+    }
```

Comment:
This logs a warning if the provider doesn't implement state, but implements init.